data_model/src/expression/operation.rs

```diff
@@ -0,0 +1,81 @@
+//! Module containing operations and their priorities.
+
+/// Type of expression operation.
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Operation {
+    MethodCall,
+    RaiseTo,
+    Multiply,
+    Divide,
+    Mod,
+    Add,
+    Subtract,
+    Greater,
+    Less,
+    Equal,
+    Not,
+    And,
+    Or,
+    Other,
+}
+
+/// Priority of operation.
+///
+/// [`First`](Operation::First) is the highest priority
+/// and [`Eight`](Operation::Eight) is the lowest.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Priority {
+    First = 1,
```

Comment:
You can explicitly set discriminant only to the first element, check this [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1f327be47cc951e08ab9b155ea0ebfac).
We can even omit setting discriminant at all rust automatically start from 0 and increment with every element.

Reply:
Oh, didn't know rust automatically starts from 0, good